src/Avalonia.Controls/TextBox.cs

```diff
@@ -446,6 +458,28 @@ public bool CanPaste
             private set { SetAndRaise(CanPasteProperty, ref _canPaste, value); }
         }
 
+        /// <summary>
+        /// Property for determining whether undo/redo is enabled
+        /// </summary>
+        public bool IsUndoEnabled
+        {
+            get { return GetValue(IsUndoEnabledProperty); }
+            set { SetValue(IsUndoEnabledProperty, value); }
+        }
+
+        public int UndoLimit
+        {
+            get { return _undoRedoHelper.Limit; }
+            set
```

Comment:
Without SetAndRaise this setter won't invoke property changed events of UndoLimitProperty.
In case if somebody has bindings to it.

Reply:
I had to use `RaisePropertyChanged` because a property or indexer may not be passed as an out or ref parameter (aka we can't use `SetAndRaise` with `_undoRedoHelper.Limit`). So I just used `RaisePropertyChanged` instead.